src/components/Input/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FC, useState, KeyboardEvent } from "react";
+
+import { SearchInputProps } from "./interface";
+
+import SuggestionsListComponent from "./suggestionsListComponent";
+import {
+  Form,
+  Input,
+  Label,
+  SearchIcon,
+  RemoveIcon,
+  FormWrapper,
+  Button,
+} from "./styledComponents";
+
+const SearchInput: FC<SearchInputProps> = ({ suggestions }) => {
+  const [filteredSuggestions, setFilteredSuggestions] = useState<string[]>([]);
+  const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(0);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [showClearButton, setShowClearButton] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const userInput = e.currentTarget.value;
+    const filteredList = suggestions.filter(
+      (suggestion) =>
+        suggestion.toLowerCase().indexOf(userInput.toLowerCase()) > -1
+    );
+
+    setSearchTerm(e.currentTarget.value);
+    setFilteredSuggestions(filteredList);
+    setActiveSuggestionIndex(0);
+    setShowSuggestions(true);
+    setShowClearButton(true);
+  };
+
+  const clearInput = () => {
+    setSearchTerm("");
+    setShowClearButton(false);
+  };
+
+  const chooseSuggestion = (e: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+    e.preventDefault();
+    setFilteredSuggestions([]);
+    setSearchTerm(e.currentTarget.innerText);
+    setActiveSuggestionIndex(0);
+    setShowSuggestions(false);
+  };
+
+  //It's working, but I would like to prevent behaviour that
+  //after I hit the last suggestion the background style don't vanish
+
+  const onKeyDown = (e: KeyboardEvent) => {
+    if (e.key === "Enter") {
+      if (activeSuggestionIndex !== -1) {
+        e.preventDefault();
+        setSearchTerm(filteredSuggestions[activeSuggestionIndex]);
+        setFilteredSuggestions([]);
+        setActiveSuggestionIndex(0);
+        setShowSuggestions(false);
+      }
+    } else if (e.key === "ArrowUp") {
+      if (activeSuggestionIndex >= 0) {
+        e.preventDefault();
+        setActiveSuggestionIndex(activeSuggestionIndex - 1);
+      }
+    } else if (e.key === "ArrowDown") {
+      if (activeSuggestionIndex < filteredSuggestions.length) {
+        e.preventDefault();
+        setActiveSuggestionIndex(activeSuggestionIndex + 1);
+      }
+    }
+  };
+
+  return (
+    <FormWrapper>
```

Comment:
We need to restructure an html a bit to comply to ARIA and HTML standards:
https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html
We could have a div named e.g. `Combobox` and its properties are as follows:
```
<Combobox
      id="combobox"
      aria-expanded="false"
      aria-owns="autocomplete-options"
      aria-haspopup="listbox"
>    
```